specs/Deployment/DeploymentAPI.md

```diff
@@ -129,6 +129,53 @@ again.
     }
 ```
 
+### DeploymentOptions
+
+#### ForceTargetApplicationShutDown
+
+When this API may install newer version of one or more of the packages (i.e. an update) and
+if there are any currently running process(es) associated with that package(s) (in other words,
+if any of the packages to be updated are currently in use), then this API will fail installing
+that package. But this update to the packages can be forced by using DeploymentOptions object
+and setting ForceTargetApplicationShutDown option before passing it to this API. This option
+when set will shutdown the application(s) associated with the package, update the package
+and restart the application(s).
+
+[ToDo] If this option is set when updating main package, then whether the out of process
+com server from the main package such as push Notifications needs to be explictly restarted
+needs to be understood. If restart it is needed, then the API will handle explicitly restarting it.
+
+When the API is updating framework package and ForceTargetApplicationShutDown option is set,
+then all dependent packages that are NOT currently in use will be immediately re-installed
+to refer to the updated framework package and all dependent packages that are currently in use
+will be re-installed, after they shut down, to refer to the updated framework package.
+
+```C#
+    if (DeploymentManager.GetStatus().Status != DeploymentStatus.Ok)
+    {
+        // Create DeploymentOptions object and set ForceTargetApplicationShutDown option
+        // to allow for force updating of the packages even if they are currently in use.
+        DeploymentOptions deploymentOptions = new DeploymentOptions {};
```

Comment:
`...new DeploymentOptions();`

Been writing a lot of C++ lately? ;-)

Reply:
you noticed that ;)